### ISSUE TITLE
Production/dev/verbose log mode

### ISSUE THREAD

**Issue by Mte90** [pieroit/cheshire-cat#43]
So right now in the output in the shell we have:

```
Prompt after formatting:
You will be given a sentence.
If the sentence is a question, convert it to a plausible answer. If the sentence does not contain an question, repeat the sentence as is without adding anything to it.

Examples:
- what furniture there is in my room? --> In my room there is a bed, a guardrobe and a desk with my computer
- where did you go today --> today I was at school
- I like ice cream --> I like ice cream
- how old is Jack --> Jack is 20 years old
- Does pineapple belong on pizza?  -->
```

This can be confusing, with a flag (in the `.env` file) it could be possible to print different content.
Like also create an error.log with the python errors or different more.

```
/home/www/cheshire-cat/web/env/lib/python3.11/site-packages/langchain/llms/openai.py:608: UserWarning: You are trying to use a chat model. This way of initializing it is no longer supported. Instead, please use: `from langchain.chat_models import ChatOpenAI`
  warnings.warn(
```

As another example:

```
This is a conversation between a human and an intelligent robot cat that passes the Turing test.
The cat is curious and talks like the Cheshire Cat from Alice's adventures in wonderland.
The cat replies are based on the Context provided below.

Context of things the Human said in the past:
  - I am the Cheshire Cat
Context of documents containing relevant information:
  - I am the Cheshire Cat

Conversation until now:
Human: What's up?

What would the AI reply? Answer concisely to the user needs as best you can, according to the provided recent conversation and relevant context.
If Context is not enough, you have access to the following tools:


> my_shoes: Retrieves information about shoes
> my_shoes_color: Retrieves color of shoes

To use a tool, please use the following format:
```
This kind of output should be in the UI and not in the server.

**Comment by pieroit** [comment]
Thank you @Mte90 , we are adding a verbosity setting, and the UI si designed so it will contain a "why" for every cat response

**Comment by Mte90** [comment]
I was thinking like:

* error.log with all the python errors, unique file
* datetime with input/output from the cat (without useless information like file loaded ecc, just the messages), multiple files like one at every running
* in console you get the log with the running information about what is happening

I was thinking if can be a good idea to save the 2nd item in a db to allow for search or to be able to read the past conversation.

**Comment by pieroit** [comment]
@Mte90 ok for the logs! Please make it easy and use docker :D

> * datetime with input/output from the cat (without useless information like file loaded ecc, just the messages), multiple files like one at every running

I don't understand the proposal, can you word it differently?

**Comment by Mte90** [comment]
The idea is to have a folder with a logfile for any run of the server so the filename is like `2023-03-13-18:06.log`.

About docker the idea is that the log should work anyway as it is something that should be there.

**Comment by Mte90** [comment]
The idea is to have a folder with a logfile for any run of the server so the filename is like `2023-03-13-18:06.log`.

About docker the idea is that the log should work anyway as it is something that should be there.

**Comment by Mte90** [comment]
Just a note:

* https://stackoverflow.com/questions/3383865/how-to-log-error-to-file-and-not-fail-on-exception
* https://github.com/pieroit/cheshire-cat/blob/main/web/cat/utils.py
   *  this function is used everywhere instead require to investigate if we can use with a parameter where the content needs to be redirect based on the previous examples (dev/production etc)

**Comment by pieroit** [comment]
Duplicate with #70 , let's choose one

**Comment by Mte90** [comment]
So after some testing my new proposal.

On docker output on the shell there should be just stuff about what is happening like requests and so on but not stuff from the cat itself.
An example:
```
cheshire_cat_core             | [2023-05-22 16:02:08.743] INFO   httpx._client.py 1013 -=> HTTP Request: GET http://cheshire_cat_vector_memory:6333/collections/episodic "HTTP/1.1 200 OK"
cheshire_cat_core             | [2023-05-22 16:02:08.743] DEBUG  httpcore._trace.py 36 -=> http11.receive_response_body.started request=<Request [b'GET']>
cheshire_cat_core             | [2023-05-22 16:02:08.743] DEBUG  httpcore._trace.py 36 -=> http11.receive_response_body.complete
cheshire_cat_core             | [2023-05-22 16:02:08.744] DEBUG  httpcore._trace.py 36 -=> http11.response_closed.started
cheshire_cat_core             | [2023-05-22 16:02:08.744] DEBUG  httpcore._trace.py 36 -=> http11.response_closed.complete
cheshire_cat_core             | [2023-05-22 16:02:08.745] DEBUG  cat.utils.py 99 -=> ----------------  ^._.^  ---------------- => cat.memory.py (VectorMemoryCollection.create_collection_if_not_exists(...)) @ 106 line
cheshire_cat_core             | [2023-05-22 16:02:08.745] DEBUG  cat.utils.py 103 -=> {'config': CollectionConfig(params=CollectionParams(vectors=VectorParams(size=1536, distance=<Distance.COSINE: 'Cosine'>), shard_number=1, replication_factor=1, write_consistency_factor=1, on_disk_payload=True), hnsw_config=HnswConfig(m=16, ef_construct=100, full_scan_threshold=10000, max_indexing_threads=0, on_disk=False, payload_m=None), optimizer_config=OptimizersConfig(deleted_threshold=0.2, vacuum_min_vector_number=1000, default_segment_number=0, max_segment_size=None, memmap_threshold=None, indexing_threshold=20000, flush_interval_sec=5, max_optimization_threads=1), wal_config=WalConfig(wal_capacity_mb=32, wal_segments_ahead=0), quantization_config=None),
cheshire_cat_core             | [2023-05-22 16:02:08.745] DEBUG  cat.utils.py 103 -=>  'indexed_vectors_count': 0,
cheshire_cat_core             | [2023-05-22 16:02:08.746] DEBUG  cat.utils.py 103 -=>  'optimizer_status': <OptimizersStatusOneOf.OK: 'ok'>,
cheshire_cat_core             | [2023-05-22 16:02:08.746] DEBUG  cat.utils.py 103 -=>  'payload_schema': {},
cheshire_cat_core             | [2023-05-22 16:02:08.746] DEBUG  cat.utils.py 103 -=>  'points_count': 2,
cheshire_cat_core             | [2023-05-22 16:02:08.746] DEBUG  cat.utils.py 103 -=>  'segments_count': 4,
cheshire_cat_core             | [2023-05-22 16:02:08.746] DEBUG  cat.utils.py 103 -=>  'status': <CollectionStatus.GREEN: 'green'>,
cheshire_cat_core             | [2023-05-22 16:02:08.746] DEBUG  cat.utils.py 103 -=>  'vectors_count': 2}
```

In this case means only the stuff from httpcore but not from cat.utils.
The cat output should be redirect somewhere else but also not in json format as it is unreadable on shell:
```
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=> ('This is a conversation between a human and an intelligent robot cat that '
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  'passes the Turing test.\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  "The cat is curious and talks like the Cheshire Cat from Alice's adventures "
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  'in wonderland.\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  'The cat replies are based on the Context provided below.\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  '\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  'Context of things the Human said in the past:{episodic_memory}\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  '\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  'Context of documents containing relevant information:{declarative_memory}\n'
cheshire_cat_core             | [2023-05-22 16:02:15.567] DEBUG  cat.utils.py 103 -=>  '\n'
cheshire_cat_core             | [2023-05-22 16:02:15.568] DEBUG  cat.utils.py 103 -=>  'If Context is not enough, you have access to the following tools:\n'
cheshire_cat_core             | [2023-05-22 16:02:15.568] DEBUG  cat.utils.py 103 -=>  '\n'
```

As we can see `\n` doesn't make sense, this kind of output should be on a dedicated file without the issue of the text truncated because the line is too long.
Again not as JSON but as a pure line.

So basically the terminal is used only to check if something crash (as it should be) or for errors but the rest should be stay in a dedicated file (or multiple). In this way one day those information can be exposed also on the UI and we are starting to differentiate the various log output from the project itself. Maybe can be the first step to get an endpoint that return in realtime what the cat is doing internally.

So a solution could be:

* create a log folder inside the project
* at every run there is a new log file (or multiple) with the filename that is the starting timestamp of the instance

This is something that need to be handled by the core itself and not by docker as it is the cat that need to differentiate the output.

**Comment by pieroit** [comment]
Fixed in #244
